### PR TITLE
Change exception type when detaching unattached sources to InvalidRequestError

### LINKED ISSUE #376: Better exception for Source.detach() on detached sources
https://github.com/stripe/stripe-python/blob/8629450495adf3ea6d2c1ae343a99c8a8480e90e/stripe/api_resources/source.py#L29

Could this get its own exception type? I want to give customers the option to remove a source from their payment methods, and right now I have to do something like this:

```py
try:
    source.detach()
except NotImplementedError:
    resync(source)
```

Catching notimpl is weird, and this isn't really the appropriate exception type I think anyway.

### PR DESCRIPTION
**Pull request by ob-stripe**
r? @brandur-stripe 
cc @stripe/api-libraries @jleclanche

Changes the exception raised when trying to detach unattached sources from `NotImplementedError` to `stripe.error.InvalidRequestError`. This is consistent with other libraries.

Fixes #376.

### CODE REVIEW

**Comment by jleclanche** [comment]
:+1:

**Comment by brandur-stripe** [comment]
Much more sensible.

LGTM.
